File: test1.js

var fs = require('fs');

var path = process.argv[2] || "/usr/share/dict/words";

var words = splitWords(fs.readFileSync(path, "utf-8"));

readStdin(function (input) {
  var inputWords = splitWords(input);
  inputWords.forEach(function (inputWord) {
    if (words.indexOf(inputWord) > -1) {
      console.log(inputWord);
    } else {
      console.log("<"+inputWord+">");
    }
  });
});

function splitWords(str) {
  return str.split(/\s+/).filter(function (w) {
    return w.length > 0;
  });
}

function readStdin(cb) {
  var data = "";
  process.stdin.setEncoding('utf8');
  process.stdin.on('readable', function() {
    var chunk = process.stdin.read();
    if (chunk !== null) {
      data += chunk;
    }
  });
  process.stdin.on('end', function() {
    cb(data);
  });
}

File: test1.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"regexp"
	"strings"
)

type whitelist []string

var wordRegexp = regexp.MustCompile("([^ \n]+)")

func readWords(path string) whitelist {
	file, err := os.Open(path)
	if err != nil {
		panic(err)
	}

	buf, err := ioutil.ReadAll(file)
	if err != nil {
		panic(err)
	}

	return strings.Split(string(buf), "\n")
}

func (w whitelist) contains(needle string) bool {
	for _, v := range w {
		if v == needle {
			return true
		}
	}
	return false
}

func main() {

	path := "/usr/share/dict/words"
	if len(os.Args) > 1 {
		path = os.Args[1]
	}

	words := readWords(path)

	input, err := ioutil.ReadAll(os.Stdin)
	if err != nil {
		panic(err)
	}

	inputWords := wordRegexp.FindAllString(string(input), -1)

	for _, w := range inputWords {
		if words.contains(w) {
			fmt.Fprintf(os.Stdout, "%s\n", w)
		} else {
			fmt.Fprintf(os.Stdout, "<%s>\n", w)
		}
	}
}
